app/adapters/whats_app_adapter/three_sixty_dialog_inbound.rb

```diff
@@ -47,7 +52,35 @@ def trigger(event, *args)
       @callbacks[event].call(*args)
     end
 
-    def initialize_sender(whats_app_message)
+    def initialize_text
+      message = whats_app_message[:messages].first
+      message[:text]&.dig(:body) || message[:button]&.dig(:text) || supported_file(message)&.dig(:caption)
+    end
+
+    def ephemeral_data?
+      return if text.blank?
+
+      request_for_more_info? || unsubscribe_text? || resubscribe_text? || request_to_receive_message?
+    end
+
+    def trigger_appropriate_callback
```

Comment:
I think `ephemeral_data?` and `trigger_appropriate_callback` can be merged. They would always be called toghether and you do the same checks again, right?

Reply:
I did that at first, but refactored it because I'd like the methods to have a single responsibility.